pkg/event/events.go

```diff
@@ -40,12 +40,20 @@ type UserEvent struct {
 
 // ImpressionEvent represents an impression event
 type ImpressionEvent struct {
-	EntityID     string `json:"entity_id"`
-	Key          string `json:"key"`
 	Attributes   []VisitorAttribute
-	VariationID  string `json:"variation_id"`
-	CampaignID   string `json:"campaign_id"`
-	ExperimentID string `json:"experiment_id"`
+	CampaignID   string           `json:"campaign_id"`
+	EntityID     string           `json:"entity_id"`
+	ExperimentID string           `json:"experiment_id"`
+	Key          string           `json:"key"`
+	Metadata     DecisionMetadata `json:"metadata"`
+	VariationID  string           `json:"variation_id"`
+}
+
+// DecisionMetadata captures additional information regarding the decision
+type DecisionMetadata struct {
+	FlagType     string `json:"flag_type"`
+	FlagKey      string `json:"flag_key"`
+	VariationKey string `json:"variation_key"`
```

Comment:
shouldn't this be defined with `omitempty` since this is an optional value?
```suggestion
	VariationKey string `json:"variation_key,omitempty"`
```